packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx

```diff
@@ -15,19 +15,75 @@ import Cancellation from "../../assets/images/cancelation.svg";
 import Whatsapp from "../../assets/images/whatsup-color-big.svg";
 import TrashIcon from "../../assets/icons/trash.svg";
 import UpcomingDonationInfo from "./UpcomingDonationInfo";
+import makeUrls, { TCalendarEvent } from "add-event-to-calendar";
+import ICalendarLink from "react-icalendar-link";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
 
 export interface UpcomingDonationScreenProps {
   bookedAppointment: BookedAppointment;
   fullName: string;
   onCancel: () => Promise<void>;
 }
 
+interface CalendarType {
+  type: string;
+  label: string;
+}
+
+const calendarTypeList: CalendarType[] = [
+  { type: "google", label: "Google" },
+  { type: "ics", label: "Apple" },
+];
+
 export default function UpcomingDonationScreen({
   fullName,
   onCancel,
   bookedAppointment,
 }: UpcomingDonationScreenProps) {
-  const donationDate = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateStart = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateEnd = new Date(eventDateStart.getTime() + 90 * 60000);
```

Comment:
appointment calendar length should be a constant (i.e. a dedicated variable to be changed easily)